From ecdbaf8e588aac54db0bd50fe876e9029b376809 Mon Sep 17 00:00:00 2001
From: ValdikSS <iam@valdikss.org.ru>
Date: Mon, 17 Sep 2018 19:04:27 +0300
Subject: [PATCH] Assume optional codecs are supported if were supported previously

This fix is required to properly follow codec reconfiguration for SBC HD
(SBC Dual Channel) in case of incoming Bluetooth connection.

Change-Id: Idb8fce75b4f628ae865fc1d3a787e34b6a29e31e
---

diff --git a/src/com/android/bluetooth/a2dp/A2dpService.java b/src/com/android/bluetooth/a2dp/A2dpService.java
index d35b206..01ecd94 100644
--- a/src/com/android/bluetooth/a2dp/A2dpService.java
+++ b/src/com/android/bluetooth/a2dp/A2dpService.java
@@ -944,11 +944,11 @@
             }
         }
         if (previousSupport == BluetoothA2dp.OPTIONAL_CODECS_SUPPORT_UNKNOWN
-                || supportsOptional != (previousSupport
-                                    == BluetoothA2dp.OPTIONAL_CODECS_SUPPORTED)) {
+                || previousSupport == BluetoothA2dp.OPTIONAL_CODECS_NOT_SUPPORTED) {
             setSupportsOptionalCodecs(device, supportsOptional);
         }
-        if (supportsOptional) {
+        if (supportsOptional
+                || previousSupport == BluetoothA2dp.OPTIONAL_CODECS_SUPPORTED) {
             int enabled = getOptionalCodecsEnabled(device);
             if (enabled == BluetoothA2dp.OPTIONAL_CODECS_PREF_ENABLED) {
                 enableOptionalCodecs(device);
